azurerm/internal/services/keyvault/key_vault_managed_hardware_security_module_resource_test.go

```diff
@@ -0,0 +1,181 @@
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return utils.Bool(resp.Properties != nil), nil
+}
+
+func (r KeyVaultManagedHardwareSecurityModuleResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_key_vault_managed_hardware_security_module" "test" {
+  name                = "kvHsm%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  sku_name            = "Standard_B1"
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  admin_object_ids    = [data.azurerm_client_config.current.object_id]
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r KeyVaultManagedHardwareSecurityModuleResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_key_vault_managed_hardware_security_module" "import" {
+  name                = azurerm_key_vault_managed_hardware_security_module.test.name
+  resource_group_name = azurerm_key_vault_managed_hardware_security_module.test.resource_group_name
+  location            = azurerm_key_vault_managed_hardware_security_module.test.location
+  sku_name            = azurerm_key_vault_managed_hardware_security_module.test.sku_name
+  tenant_id           = azurerm_key_vault_managed_hardware_security_module.test.tenant_id
+  admin_object_ids    = azurerm_key_vault_managed_hardware_security_module.test.admin_object_ids
+}
+`, r.basic(data))
+}
+
+func (r KeyVaultManagedHardwareSecurityModuleResource) purgeProtectionAndSoftDelete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_key_vault_managed_hardware_security_module" "test" {
+  name                       = "kvHsm%d"
+  resource_group_name        = azurerm_resource_group.test.name
+  location                   = azurerm_resource_group.test.location
+  sku_name                   = "Standard_B1"
+  soft_delete_retention_days = 7
+  purge_protection_enabled   = true
+  tenant_id                  = data.azurerm_client_config.current.tenant_id
+  admin_object_ids           = [data.azurerm_client_config.current.object_id]
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r KeyVaultManagedHardwareSecurityModuleResource) updateTags(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_key_vault_managed_hardware_security_module" "test" {
+  name                = "kvHsm%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = azurerm_resource_group.test.location
+  sku_name            = "Standard_B1"
+  tenant_id           = data.azurerm_client_config.current.tenant_id
+  admin_object_ids    = [data.azurerm_client_config.current.object_id]
+
+  tags = {
+    Env = "Test"
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (KeyVaultManagedHardwareSecurityModuleResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
```

Comment:
the provider block shouldn't be in the template, but instead in each test, since this could vary per test - can we fix this?